spec/unit/provider/package/yum_spec.rb

```diff
@@ -12,6 +12,40 @@
     expect(described_class.specificity).to be < 200
   end
 
+  describe "should have logical defaults" do
+    [2, 2018].each do |ver|
+      it "should be the default provider on Amazon Linux #{ver}" do
+        allow(Facter).to receive(:value).with(:operatingsystem).and_return('amazon')
+        allow(Facter).to receive(:value).with(:osfamily).and_return('redhat')
+        allow(Facter).to receive(:value).with(:operatingsystemmajrelease).and_return('2', '2018')
```

Comment:
Good catch! Even better if you use `.and_return(ver)`, so we don't have to duplicate the version we're testing!